### PR TITLE
🐛 [amp-story] don't send story-page-visible for ad pages

### LINKED ISSUE #33783: 🐞[story ads] Analytics pings on story ad pages
If a pub uses story analytics with a `story-page-visible` trigger an ad page will send this ping with the same page index as the page before the ad, leading to what appears to be more page views at pages 7 & 14 (or wherever we start placing the ads). 

This includes pubs using analytics through auto analytics.

I think there are two options
1. Don't fire the `story-page-visible` trigger when the page is an ad
2. Send something like `isAd=true` along with the trigger

I would lean towards (2) as people would likely want to keep track of when this happens and can lead to slicing into ad specific metrics in e.g. GA

Repro on any page using ads + story auto analytics e.g. https://www.inputmag.com/style/how-to-dress-look-like-lil-nas-x-under-100/amp

to/ @ampproject/wg-stories for discussion
cc/ @ampproject/wg-monetization

### PR DESCRIPTION
**Pull request by calebcordry**
Closes #33783

### CODE REVIEW

**Review by processprocess — APPROVED** [review]
Looks good based on the discussion in #33783.
I'll default to @Enriqe for final approval.

File: extensions/amp-story/1.0/test/test-story-analytics.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  it('sends story-page-visible on current page change', () => {
+    const triggerAnalyticsStub = env.sandbox.stub(
+      analytics,
+      'triggerAnalyticsEvent'
+    );
+    storeService.dispatch(Action.CHANGE_PAGE, {
+      id: 'page-1',
+      index: 0,
+    });
+    expect(triggerAnalyticsStub).calledOnceWithExactly(
```
Comment (Enriqe) [inline extensions/amp-story/1.0/test/test-story-analytics.js:41]:
```suggestion
    expect(triggerAnalyticsStub).to.have.been.calledOnceWithExactly(
```

File: extensions/amp-story/1.0/story-analytics.js [hunk 2]
```diff
@@ -122,7 +122,7 @@ export class StoryAnalyticsService {
     this.storeService_.subscribe(
       StateProperty.CURRENT_PAGE_ID,
       (pageId) => {
-        if (!pageId) {
+        if (!pageId || this.storeService_.get(StateProperty.AD_STATE)) {
```
Comment (Enriqe) [inline extensions/amp-story/1.0/story-analytics.js:125]:
```suggestion
        const isAd = this.storeService_.get(StateProperty.AD_STATE); 
        if (!pageId || isAd) {
```

**Review by Enriqe — CHANGES_REQUESTED** [review]
Thanks for the PR with tests :) just a couple of comments

**Review by gmajoulet — APPROVED** [review]

File: extensions/amp-story/1.0/test/test-story-analytics.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      el,
+      'story-page-visible',
+      env.sandbox.match({storyPageIndex: 0, storyPageId: 'page-1'})
+    );
+  });
+
+  it('does not send story-page-visible on ad page', () => {
```
[the remark this answers is not in the document]
Comment (calebcordry) [reply extensions/amp-story/1.0/test/test-story-analytics.js:48]: new test added.

**Review by Enriqe — APPROVED** [review]
🎉

**Comment by calebcordry** [comment]
Merging this one is blocked by #33993

**Pull request merged**